src/app.ts

```diff
@@ -12,15 +12,74 @@ interface TemperatureSummary {
   low: number
   average: number
 }
+/* dictionary of dictionarys for the summarys, i use a counter for  */
+interface IndexedSumary {
+  [city: string]: {
+    [time: string]: {
+      summary: TemperatureSummary
+      count: number
+    }
+  }
+}
+const temperatures: IndexedSumary = {}
 
 export function processReadings(readings: TemperatureReading[]): void {
-  // add here your code
+  //add here your code
+  /*we iterate over all the readings, if there isn't a valid entry for a key, it creates it
+  each read updates the values if it is necesary
+  
+  i used for of instead of reduce because its faster as far as I have been able to research
+  and 1 loop does less comparations than 3*/
+  for (const read of readings) {
+    const readTimeToString: string = read.time.toString()
```

Comment:
Technically you don't need to call `toString` on the date. If you use it as a key it will automatically be a string. You'll get a different result between the browser and node though. It may be safer to use the time since epoch as the key instead? That may be easier to work with than a time stamped string?

```
read.time.getTime()
```

Reply:
:o to be honest, I hadn't considered it, I tried read.time.getTime() and it works well, thanks for the suggestion

And... when I tried with the automatic parser to string I get some problems, maybe it was just my TS version or my computer, I will check that thoroughly.

Reply:
Interesting. How did you try to use the automatic date string? This is what it looks like in the chrome console:
![image](https://user-images.githubusercontent.com/17991396/123832051-2e7b0000-d8c2-11eb-8061-9d9f0f1863e3.png)